src/Trie.mo

```diff
@@ -1517,12 +1521,57 @@ module {
   /// ```motoko include=initialize
   /// trie := Trie.put(trie, key "hello", Text.equal, 42).0;
   /// trie := Trie.put(trie, key "bye", Text.equal, 32).0;
-  /// // remove the value associated with "hello"
+  /// // remove the entry associated with "hello"
   /// trie := Trie.remove(trie, key "hello", Text.equal).0;
   /// assert (Trie.get(trie, key "hello", Text.equal) == null);
   /// ```
-  public func remove<K, V>(t : Trie<K, V>, k : Key<K>, k_eq : (K, K) -> Bool) : (Trie<K, V>, ?V) {
-    replace(t, k, k_eq, null)
+  public func remove<K, V>(trie : Trie<K, V>, key : Key<K>, equal : (K, K) -> Bool) : (Trie<K, V>, ?V) {
+    func combine(left : Trie<K, V>, right : Trie<K, V>) : Trie<K, V> {
```

Comment:
Drive-by: Try to minimise the use of nested functions, since they imply a new closure allocation for each call of `remove`. If they do not actually use anything from the outer function, like here, it's better to turn them into private functions of the module.

Reply:
Thank you very much for this suggestion. I will improve this.